app/services/articles/creator.rb

```diff
@@ -14,12 +14,11 @@ def call
       raise if RateLimitChecker.new(user).limit_by_action("published_article_creation")
 
       article = save_article
-
       if article.persisted?
         NotificationSubscription.create(user: user, notifiable_id: article.id, notifiable_type: "Article", config: "all_comments")
-        Notification.send_to_followers(article, "Published") if article.published
+        Notification.send_to_followers(article, "Published") if article.published?
```

Comment:
I wonder if we shouldn't move this check inside the methods (send to followers and the dispatch event), the guard if for such an important detail means we're one copy and paste away from forgetting that part and sending private data instead.

If we centralize the check it should be easier to make sure it never happens.

Reply:
I don't want to change `send_to_followers` as a part of this pr (though I touched it a bit, oops).
`send_to_followers` could be potentially used with other types of objects (though as I see after a brief look currently it isn't), and we need to change `_without_delay` as well then. Anyway, I think it's a task for another pr.

As for the check for the `dispatch_event`, I moved it a bit. Maybe it's not that you suggested.
I don't think it's a good idea to move the logic inside the `DispatchEvent` (if you meant that), it's not the dispatcher responsibility to check the object, and later we will send other types of objects/unpublished articles. Maybe it could be another class' responsibility (like we did with the adapter), but not at this moment.

Reply:
I think you took the right decision. I agree with both assesments, we'll get to the right way to do this incrementally, in the meantime we have tests :)